### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by FilbertKimasa**
In this PR, I created a basic user interface to:

- [ ] List all people
- [ ] Create a person 
- [ ] Create a book
- [ ] Create a rental
- [ ] List all books
- [ ] List all rentals for a given person id

.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,40 +1,87 @@
...
+  print 'Enter your choice: '
+end
+
+def take_user_choice
+  gets.chomp.to_i
+end
+
+def create_person(app)
+  print 'Enter person name: '
+  name = gets.chomp
```
Comment (FrederickMih) [inline main.rb]: - [ ] Hi @FilbertKimasa, let's once more kindly watch the video with the UI example in the project's requirements. It is always important to adhere to our project's requirements. We needs to first of all request the user to select whether to create a teacher or a student as in the video :+1: This also makes the UI looks engaging.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each { |person| puts "#{person.name} (ID: #{person.id})" }
+  end
+
+  def create_person(name, age, type)
+    if type == 'teacher'
+      teacher = Teacher.new(nil, age, name: name)
```
Comment (FrederickMih) [inline app.rb]: - [ ] Once more kindly watch the video again. The arguments required to create a teacher are name, age, and specialization.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      @people << teacher
+      puts "#{teacher.name} (Teacher) created with ID: #{teacher.id}"
+    elsif type == 'student'
+      student = Student.new(nil, age, name: name)
```
Comment (FrederickMih) [inline app.rb]: - [ ] Permission is required to create a student of age < 18. Kindly stick to the requirements.

File: main.rb [hunk 4]
```diff
@@ -1,40 +1,87 @@
...
+  title = gets.chomp
+  print 'Enter book author: '
+  author = gets.chomp
+  app.create_book(title, author)
+end
+
+def create_rental(app)
+  print 'Enter person ID: '
+  person_id = gets.chomp.to_i
+  print 'Enter book title: '
```
Comment (FrederickMih) [inline main.rb]: - [ ] As in the video, let's kindly request the user to first of all select the book to rent out and next, the person to rent out the book by number(Not ID), then lastly the date to rent out the book :+1:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @FilbertKimasa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- The main method was correctly invoked at the end of the main.rb file :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (Whoistolu) [reply main.rb]: - Well resolved

File: app.rb (hunk 2 above)
Comment (Whoistolu) [reply app.rb]: - Well resolved

File: main.rb (hunk 4 above)
Comment (Whoistolu) [reply main.rb]: - Well resolved

File: app.rb (hunk 3 above)
Comment (Whoistolu) [reply app.rb]: - Well resolved

**Review by Whoistolu — APPROVED** [review]
### Hi @FilbertKimasa 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**